source/core/shape.cc

```diff
@@ -0,0 +1,427 @@
+  // distance calculations
+  Point<dim> center_of_rotation = position;
+
+  Point<dim> rotated_point;
+  Point<dim> translated_point;
+
+  // Rotation from the solid orientation
+  // Angular position around x, y and z axis
+  Tensor<1, 3> theta = orientation;
+  Point<dim>   centralized_point;
+  centralized_point              = evaluation_point - center_of_rotation;
+  Point<dim> centralized_rotated = centralized_point;
+  // Selection of the first axis around which to rotate:
+  // x -> 0, y -> 1, z -> 2
+  // In 2D, only rotation around the z axis is possible
+  if (dim == 2)
+    {
+      if (std::abs(theta[2]) > 1e-10)
+        {
+          Tensor<2, 2> rotation_matrix =
+            Physics::Transformations::Rotations::rotation_matrix_2d(theta[2]);
+
+          // Multiplication
+          centralized_rotated.clear();
+          for (unsigned int j = 0; j < dim; ++j)
+            {
+              for (unsigned int k = 0; k < dim; ++k)
+                {
+                  centralized_rotated[j] +=
+                    rotation_matrix[j][k] * centralized_point[k];
+                }
+            }
+          centralized_point = centralized_rotated;
+        }
+    }
+  else if (dim == 3)
+    {
+      for (unsigned int i = 0; i < 3; ++i)
+        {
+          if (std::abs(theta[i]) > 1e-10)
+            {
+              Point<3> axis;
+              axis[i] = 1.;
+              Tensor<2, 3> rotation_matrix =
+                Physics::Transformations::Rotations::rotation_matrix_3d(
+                  axis, theta[i]);
+
+              // Multiplication
+              centralized_rotated.clear();
+              for (unsigned int j = 0; j < dim; ++j)
+                {
+                  for (unsigned int k = 0; k < dim; ++k)
+                    {
+                      centralized_rotated[j] +=
+                        rotation_matrix[j][k] * centralized_point[k];
+                    }
+                }
+              centralized_point = centralized_rotated;
+            }
+        }
+    }
+  rotated_point = centralized_rotated + center_of_rotation;
+
+  // Translation from the solid position
+  translated_point = rotated_point - position;
+
+  return translated_point;
+}
+
+template <int dim>
+double
+Sphere<dim>::value(const Point<dim> & evaluation_point,
+                   const unsigned int component) const
+{
+#if (DEAL_II_VERSION_MAJOR < 10 && DEAL_II_VERSION_MINOR < 4)
+  return p.distance(this->position) - this->effective_radius;
+#else
+  Functions::SignedDistance::Sphere<dim> sphere_function(
+    this->position, this->effective_radius);
+  return sphere_function.value(evaluation_point);
+#endif
+}
+
+template <int dim>
+std::shared_ptr<Shape<dim>>
+Sphere<dim>::static_copy() const
+{
+  std::shared_ptr<Shape<dim>> copy =
+    std::make_shared<Sphere<dim>>(this->effective_radius);
+  copy->position    = this->position;
+  copy->orientation = this->orientation;
```

Comment:
I don't underestand. Why does the constructor has only the effective radius? The position AND the orientation should also be in the constructor instead of being set  publicly like this.